pkg/codegen/dotnet/gen_program.go

```diff
@@ -280,7 +280,12 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 		if langInfo, found := p.Language["csharp"]; found {
 			csharpInfo, ok := langInfo.(CSharpPackageInfo)
 			if ok {
-				namespace := namespaceName(csharpInfo.Namespaces, p.Name)
+				pkgName := p.Name
+				if name := csharpInfo.PackageName; name != "" {
```

Comment:
Note that the changes in `gen_program.go` are for _program_ gen, i.e. generating a .NET program from PCL.

If you want to support this for _sdk_ gen, you'll also need to make changes to https://github.com/pulumi/pulumi/blob/master/pkg/codegen/dotnet/gen.go

And additionally, changes to the _docs_ gen, if necessary: https://github.com/pulumi/pulumi/blob/master/pkg/codegen/dotnet/doc.go